examples/forward_simulation.rs

```diff
@@ -669,16 +669,13 @@ fn main() {
         &mut tables,
     );
 
-    match simplify.is_some() {
-        true => (),
-        false => {
-            let _ = tskit_tables.build_index().unwrap();
-        }
-    };
-
     add_tskit_mutation_site_tables(&tables, &origin_times, g.into(), &mut tskit_tables);
 
-    tskit_tables
-        .dump(&outfile, tskit::TableOutputOptions::default())
+    let ts = match tskit_tables.tree_sequence(tskit::TreeSequenceFlags::BUILD_INDEXES) {
```

Comment:
This doesn't allow us to dump un-simplified tables.  We need tests for this, too.